### PR TITLE
Delay pmon non-critical daemon

### PR DESCRIPTION
**Pull request by yuazhe**
#### Why I did it
After this pull request https://github.com/sonic-net/sonic-buildimage/pull/19190 , the pmon has been added to the start list in fast/warm reboot scenarios. However, certain non-critical daemons of pmon could be delayed, resulting in a saving of approximately 1 second in the reboot process. For performance considerations, especially as the current time usage of fast reboot is closer to 30 seconds limitation, this change could ease the pressure.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
add a script as fast/warm reboot monitor and relative supervisord rlues.
once the script exited means the reboot process has ended, other delayed daemon would then initialize.

#### How to verify it
check the fast/warm reboot time usage

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [x] 202311
- [ ]

### CODE REVIEW

**Review by keboliu — APPROVED** [review]

**Comment by keboliu** [comment]
Hi @prgeor would you please review?

File: dockers/docker-platform-monitor/docker-pmon.supervisord.conf.j2
Comment (prgeor) [inline dockers/docker-platform-monitor/docker-pmon.supervisord.conf.j2:1]: @yuazhe Can you remove python2 check like these?
```
command={% if API_VERSION == 3 and 'syseepromd' not in python2_daemons %}python3 {% else %} python2 {% endif %}/usr/local/bin/syseepromd
```

Comment (yuazhe) [reply dockers/docker-platform-monitor/docker-pmon.supervisord.conf.j2:1]: @prgeor why should I do that, this is not in this pr's scope?

**Review by prgeor — APPROVED** [review]

**Comment by liat-grozovik** [comment]
@yuazhe please handle conflict so we can merge.
if this cannot be cleanly taken to 202405 please raise a new PR and tagged it so we can take to both branches

**Comment by yuazhe** [comment]
@liat-grozovik conflict had been resolved.

**Comment by liat-grozovik** [comment]
@prgeor any comment or we can go a head and merge?

**Review by liat-grozovik — APPROVED** [review]

**Comment by bingwang-ms** [comment]
@prgeor Can you help review again?

File: dockers/docker-platform-monitor/docker-pmon.supervisord.conf.j2 [hunk 1]
```diff
@@ -28,6 +28,16 @@ stdout_logfile=syslog
 stderr_logfile=syslog
 dependent_startup=true
 
+
+{% if delay_non_critical_daemon %}
+[program:delay]
+command=python3 /usr/bin/delay.py
```
Comment (prgeor) [inline dockers/docker-platform-monitor/docker-pmon.supervisord.conf.j2:34]: @yuazhe please use normal sleep...not sure what so specific about delay.py

**Review by prgeor — APPROVED** [review]

**Pull request merged**